### PR TITLE
refactor: introduce possibility to observe host for named slot changes

### PR DESCRIPTION
**Pull request by kyubisation**
We have various use cases where we want to remove or render an entire section of elements, depending on whether a specific element is slotted into a named slot.
This PR introduces functionality to define a named slot state and a way to observe the light DOM of the host for changes.

### CODE REVIEW

**Review by DavideMininni-Fincons — COMMENTED** [review]
It looks great. 🚀 
Question: we can't test it until the Stencil slotchange bug is fixed, right?

**Comment by jeripeierSBB** [comment]
> It looks great. 🚀 Question: we can't test it until the Stencil slotchange bug is fixed, right?

Maybe with an e2e test?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Generaly looks super :)

should we add documentation in this Pr?

**Comment by jeripeierSBB** [comment]
Reminder: Please use logic in sbb-link-list

File: src/global/helpers/observe-named-slot-changes.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    (map, mutation) =>
+      map.set(
+        mutation.target,
+        extractSlotNames([mutation.addedNodes, mutation.removedNodes], map.get(mutation.target))
+      ),
+    new Map<Node, Set<string>>()
+  );
+  mutationMap.forEach((slotNames, host) => {
+    if (slotNames.size) {
+      host.dispatchEvent(new CustomEvent('sbbSlotNameChange', { detail: slotNames }));
```
Comment (kyubisation) [inline src/global/helpers/observe-named-slot-changes.ts:29]: Better name for event and utility functions

File: src/components/sbb-form-field/sbb-form-field.tsx [hunk 2]
```diff
@@ -168,24 +192,16 @@ export class SbbFormField {
...
-                  <span aria-hidden="true">&nbsp;{i18nOptional[this._currentLanguage]}</span>
-                )}
-              </label>
+              {(this.label || this._namedSlots.label) && (
+                <label class="form-field__label" htmlFor={this._input?.id}>
+                  <slot name="label">
+                    <span>{this.label}</span>
+                  </slot>
+                  {this.optional && (
+                    <span aria-hidden="true">&nbsp;{i18nOptional[this._currentLanguage]}</span>
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/sbb-form-field/sbb-form-field.tsx:201]: Good question. Technically a form element can declare itself, whether it is required or not (via required attribute).

**Pull request merged**